antismash/common/secmet/locations.py

```diff
@@ -16,59 +16,59 @@
     UnknownPosition,
 )
 
+# extend FeatureLocation and CompoundLocation with bio_start / bio_end until upstream implementation
+FeatureLocation.bio_start = property(lambda self: self.end if self.strand == -1 else self.start)
+FeatureLocation.bio_end = property(lambda self: self.start if self.strand == -1 else self.end)
+CompoundLocation.bio_start = property(lambda self: self.parts[0].bio_start)
+CompoundLocation.bio_end = property(lambda self: self.parts[-1].bio_end)
 
-def convert_protein_position_to_dna(start: int, end: int, location: FeatureLocation) -> Tuple[int, int]:
+
+def convert_protein_position_to_dna(start: int, end: int, location: FeatureLocation, codon_start: int) -> Tuple[int, int]:
```

Comment:
I agree, this would have been a next step. But it would work even without changes in translation logic. You asked for a smaller commits, i provided one. Still learning, please have patience.